api/services/auth.js

```diff
@@ -1,35 +1,111 @@
-const jwt = require('jsonwebtoken');
-const bcrypt = require('bcrypt');
-
-const models = require('../../database');
-
-const generateToken = payload => {
-  const user = {
-    id: payload.id,
-    email: payload.email,
-  };
-  const token = jwt.sign(user, process.env.SECRET, {
-    expiresIn: process.env.EXPIRES_IN,
-  });
-  return { token: token };
-};
-
-const validPassword = (password, hash) => {
-  return bcrypt.compareSync(password, hash);
-};
+const { hashHelpers, jwtHelpers } = require('../helpers');
+const { ErrorHandler } = require('../middlewares/errorHandler');
+const db = require('../../database');
 
 const login = async data => {
   const email = data.email;
   const password = data.password;
-  const user = await models.Users.findOne({ where: { email } });
+  const user = await db.Users.findOne({ where: { email } });
   if (!user) throw Error('User not found.');
-  if (!validPassword(password, user.password)) {
+  if (!hashHelpers.validPassword(password, user.password)) {
     throw Error('Wrong password.');
   }
 
-  return generateToken(user.dataValues);
+  return jwtHelpers.generateToken(user.dataValues);
 };
 
+/**
+ * TODO: It needs to rebuild process of creating user Instance with Transactions
+ *  https://sequelize.org/master/manual/transactions.html
+ */
+
+async function regInDataBaseUser({ firstName, lastName, email, password }) {
```

Comment:
We have function "login", so maybe you should rename this function to "register" or "signUp" or smth.

Reply:
done